apollo-federation/src/operation/mod.rs

```diff
@@ -539,6 +547,73 @@ impl HasSelectionKey for Selection {
     }
 }
 
+impl Ord for Selection {
+    fn cmp(&self, other: &Self) -> Ordering {
+        fn compare_directives(d1: &DirectiveList, d2: &DirectiveList) -> Ordering {
+            if d1 == d2 {
+                Ordering::Equal
+            } else if d1.is_empty() {
+                Ordering::Less
+            } else if d2.is_empty() {
+                Ordering::Greater
+            } else {
+                d1.to_string().cmp(&d2.to_string())
```

Comment:
This could be premature optimization, so take it with a grain of salt, but we should be able to avoid construct two strings. Both directive lists are internally sorted, so we should be able to iterate over the two lists, do some kind of comparison (effectively a series of string comps), and `find_map` the first non-equal directives. This also prevents us from doing the direct equals check above (which is largely what I just described, a bunch of string comps).

All that said, I don't know if they would have a measurable affect on performance.

Reply:
yeah I wouldn't bother for now as this seems like a very rare scenario, i.e. we have side by side two inline fragments on the same type but with different directives, i.e.

```graphql
... on A @foo @bar { .. }
... on A @bar @baz { .. }
```

If we see this as a problem we can definitely make it smarter, but I'd just keep it as is for now.